source/routing/raptor_api.cpp

```diff
@@ -49,7 +49,129 @@ www.navitia.io
 namespace navitia { namespace routing {
 
 // Usefull to protect raptor loop
-static const uint max_nb_raptor_call = 100;
+static const uint MAX_NB_RAPTOR_CALL = 100;
+
+/**
+ * @brief This function determine the breaking condition that depends on nb of found journeys,
+ * min_nb_journeys and time frame.
+ *
+ * @details
+ *
+ *   requested_datetime           requested_datetime + timeframe_duration
+ *   -------[--------------------------------------------]-------------------------->
+ *                  minimal period to explore
+ *
+ * timeframe_limit (request_date_secs + timeframe_duration):
+ *          The end datetime of original request's timeframe (we have to explore at minimum this period)
+ *
+ *                   min_nb_journeys is set?
+ *                    /               \
+ *                   / Y               \ N
+ *          timeframe is set?        timeframe is set?
+ *            /       \                  /     \
+ *           / Y       \ N              / Y      \ N
+ *       Case 1      Case 2          Case 3     Case 4
+ *
+ *
+ */
+static bool keep_going(const uint32_t total_nb_journeys,
```

Comment:
We have to consider max_nb_journeys here also (we absolutely want to respect it, so there's no use to keep going if we are over it already)